pandas/core/base.py

```diff
@@ -653,7 +655,17 @@ class IndexOpsMixin:
 
     # ndarray compatibility
     __array_priority__ = 1000
-    _deprecations = frozenset(["item"])
+    _deprecations = frozenset(
+        [
+            "tolist",  # tolist is not deprecated, just suppressed in the __dir__
+            "base",
+            "data",
+            "item",
+            "itemsize",
+            "flags",
+            "strides",
+        ]
+    )  # type: FrozenSet[str]
```

Comment:
You might not need this annotation; I think mypy should be able to infer

Reply:
Tried again, and mypy complains without the annotation. My guess is that it cannot be sure that the added set should only contain strings.